impeller/toolkit/interop/backend/vulkan/context_vk.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/toolkit/interop/backend/vulkan/context_vk.h"
+
+#include "flutter/fml/paths.h"
+#include "impeller/entity/vk/entity_shaders_vk.h"
+#include "impeller/entity/vk/framebuffer_blend_shaders_vk.h"
+#include "impeller/entity/vk/modern_shaders_vk.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/vk/compute_shaders_vk.h"
+
+namespace impeller::interop {
+
+static std::vector<std::shared_ptr<fml::Mapping>>
+CreateShaderLibraryMappings() {
+  return {
+      std::make_shared<fml::NonOwnedMapping>(impeller_entity_shaders_vk_data,
+                                             impeller_entity_shaders_vk_length),
+      std::make_shared<fml::NonOwnedMapping>(impeller_modern_shaders_vk_data,
+                                             impeller_modern_shaders_vk_length),
+      std::make_shared<fml::NonOwnedMapping>(
+          impeller_framebuffer_blend_shaders_vk_data,
+          impeller_framebuffer_blend_shaders_vk_length),
+      std::make_shared<fml::NonOwnedMapping>(
+          impeller_compute_shaders_vk_data, impeller_compute_shaders_vk_length),
+  };
+}
+
+ScopedObject<Context> ContextVK::Create(const Settings& settings) {
+  if (!settings.IsValid()) {
+    VALIDATION_LOG << "Invalid settings for Vulkan context creation.";
+    return {};
+  }
+  impeller::ContextVK::Settings impeller_settings;
+  impeller_settings.shader_libraries_data = CreateShaderLibraryMappings();
+  impeller_settings.cache_directory = fml::paths::GetCachesDirectory();
+  impeller_settings.enable_validation = true;
```

Comment:
should this be reading from the settings?

Reply:
Nice spot. This gave me so much trouble since reading from the settings was what I was doing initially. But we set setting to this [to `true` here](https://github.com/flutter/engine/blob/5fb04a625bac45f89289f28e69291fd822c36ec0/impeller/playground/playground_impl.cc#L44). This causes the system global proc. table to be initialized twice. And, depending on whether the second instance enabled or disables validations, proc. entries are zeroed out. So teardown of say the debug printers in the second context to be torn down won't find the proc. table entry for the deleter. And cause a crash.

Reply:
Oh, but you mentioned settings. While I was using switches (that the settings are inferred from). Perhaps I could steal the underlying contexts settings directly. Momentido.